src/AzureClient/AzureClient.cs

```diff
@@ -1,24 +1,249 @@
+            }
+
+            var credentials = new Rest.TokenCredentials(AuthenticationResult.AccessToken);
+
+            QuantumClient = new QuantumClient(credentials);
+            QuantumClient.SubscriptionId = subscriptionId;
+            QuantumClient.ResourceGroupName = resourceGroupName;
+            QuantumClient.WorkspaceName = workspaceName;
+
+            ActiveWorkspace = new Azure.Quantum.Workspace(
+                QuantumClient.SubscriptionId, QuantumClient.ResourceGroupName,
+                QuantumClient.WorkspaceName, AuthenticationResult?.AccessToken);
+
+            try
+            {
+                var jobsList = await QuantumClient.Jobs.ListAsync();
+                channel.Stdout($"Found {jobsList.Count()} jobs in Azure Quantum workspace {workspaceName}");
+            }
+            catch (Exception e)
+            {
+                channel.Stderr(e.ToString());
+                return AzureClientError.WorkspaceNotFound.ToExecutionResult();
+            }
+
+            return QuantumClient.ToJupyterTable().ToExecutionResult();
+        }
+
+        /// <inheritdoc/>
+        public async Task<ExecutionResult> PrintConnectionStatusAsync(IChannel channel)
+        {
+            if (QuantumClient == null)
+            {
+                return AzureClientError.NotConnected.ToExecutionResult();
+            }
+
+            return QuantumClient.ToJupyterTable().ToExecutionResult();
+        }
+
+        /// <inheritdoc/>
+        public async Task<ExecutionResult> SubmitJobAsync(
+            IChannel channel,
+            IOperationResolver operationResolver,
+            string operationName)
+        {
+            if (ActiveWorkspace == null)
+            {
+                channel.Stderr("Must call %connect before submitting a job.");
+                return AzureClientError.NotConnected.ToExecutionResult();
+            }
+
+            if (ActiveTargetName == null)
+            {
+                channel.Stderr("Must call %target before submitting a job.");
+                return AzureClientError.NoTarget.ToExecutionResult();
+            }
+
+            if (string.IsNullOrEmpty(operationName))
+            {
+                channel.Stderr("Must pass a valid Q# operation name to %submit.");
+                return AzureClientError.NoOperationName.ToExecutionResult();
+            }
+
+            var operationInfo = operationResolver.Resolve(operationName);
+            var entryPointInfo = new EntryPointInfo<QVoid, Result>(operationInfo.RoslynType);
+            var entryPointInput = QVoid.Instance;
+            var machine = Azure.Quantum.QuantumMachineFactory.CreateMachine(ActiveWorkspace, ActiveTargetName, ConnectionString);
+            if (machine == null)
+            {
+                channel.Stderr($"Could not find an execution target for target {ActiveTargetName}.");
+                return AzureClientError.NoTarget.ToExecutionResult();
```

Comment:
Just to double-check, should these calls to `ToExecutionResult` have an `ExecuteStatus.Error` parameter so as to raise the appropriate error in the client?
```suggestion
                return AzureClientError.NoTarget.ToExecutionResult(ExecuteStatus.Error);
```

Reply:
The `AzureClientError.ToExecutionResult` extension function automatically sets `ExecuteStatus.Error` in the `ExecutionResult` object. I think that makes sense, since an `AzureClientError` is always an error?

---
In reply to: [427470175](https://github.com/microsoft/iqsharp/pull/144#discussion_r427470175) [](ancestors = 427470175)

Reply:
That make sense, I had just missed that overriding of `ToExecutionResult()this object)`. Thanks for the explanation!